Mage.Sets/src/mage/cards/k/KharnTheBetrayer.java

```diff
@@ -0,0 +1,142 @@
+        ability.addEffect(new BlocksIfAbleSourceEffect(Duration.WhileOnBattlefield).setText("blocks each combat if able").concatBy("or"));
+        this.addAbility(ability.withFlavorWord("Berzerker"));
+
+        // When you lose control of {this}, draw two cards
+        this.addAbility(new KharnTheBetrayerTriggeredAbility().withFlavorWord("Sigil of Corruption"));
+
+        // If damage would be dealt to {this}, prevent that damage and an opponent of your choice gains control of it.
+        Ability replacementAbility = new SimpleStaticAbility(new KharnTheBetrayerEffect());
+        this.addAbility(replacementAbility.withFlavorWord("The Betrayer"));
+    }
+
+    protected KharnTheBetrayer(final KharnTheBetrayer card) {
+        super(card);
+    }
+
+    @Override
+    public Card copy() {
+        return new KharnTheBetrayer(this);
+    }
+    
+}
+
+class KharnTheBetrayerEffect extends ReplacementEffectImpl {
+
+    public KharnTheBetrayerEffect() {
+        super(Duration.WhileOnBattlefield, Outcome.PreventDamage);
+        this.staticText = "If damage would be dealt to {this}, prevent that damage and an opponent of your choice gains control of it.";
+    }
+
+    protected KharnTheBetrayerEffect(final KharnTheBetrayerEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public boolean replaceEvent(GameEvent event, Ability source, Game game) {
+        int damage = game.preventDamage(event, source, game, Integer.MAX_VALUE).getPreventedDamage();
+        Player player = game.getPlayer(source.getControllerId());
+
+        if (player != null && damage > 0) {
+            TargetPlayer target = new TargetOpponent();
+            target.setNotTarget(true);
+            if (!player.choose(outcome, target, source, game)) {
+                return false;
+            }
+            ContinuousEffect effect = new GainControlTargetEffect(Duration.Custom, true, target.getFirstTarget());
+            effect.setTargetPointer(new FixedTarget(source.getSourceId(), game));
+            
+            ReflexiveTriggeredAbility ability 
+                = new ReflexiveTriggeredAbility(effect, false, "an opponent of your choice gains control of it.");
+            game.fireReflexiveTriggeredAbility(ability, source);
+        }
+        return false;
+    }
+
+    @Override
+    public boolean checksEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.DAMAGE_PERMANENT;
+    }
+
+    @Override
+    public boolean applies(GameEvent event, Ability source, Game game) {
+        return event.getTargetId().equals(source.getSourceId());
+    }
+
+    @Override
+    public ContinuousEffect copy() {
+        return new KharnTheBetrayerEffect(this);
+    }
+    
+}
+
+class KharnTheBetrayerTriggeredAbility extends TriggeredAbilityImpl {
+
+    KharnTheBetrayerTriggeredAbility () {
+        super(Zone.BATTLEFIELD, new DrawCardSourceControllerEffect(2));
```

Comment:
DrawCardSourceControllerEffect is not the proper effect, since on trigger, the source should already be the player that gained control of Kharn (Maybe I'm wrong? this needs testing.)

If this needs to be changed. I would guess having a DrawCardTargetEffect with the target set during checkTrigger right before returning true would work?